python/modules/Stop0lBaselineProducer.py

```diff
@@ -199,23 +203,24 @@ def analyze(self, event):
         PassCaloMETRatio= (met.pt / caloMET.pt ) < 5 if caloMET.pt > 0 else True
         PassEventFilter = self.PassEventFilter(flags) and PassJetID
 
-        countEle, countMu, countIsk = self.calculateNLeptons(electrons, muons, isotracks)
+        countEle, countMu, countIsk, countTauPOG = self.calculateNLeptons(electrons, muons, isotracks, taus)
         PassElecVeto    = countEle == 0
         PassMuonVeto    = countMu == 0
         PassIsoTrkVeto  = countIsk == 0
-        PassLeptonVeto  = PassElecVeto and PassMuonVeto and PassIsoTrkVeto
+	PassTauVeto	= countTauPOG == 0
+        PassLeptonVeto  = PassElecVeto and PassMuonVeto and PassIsoTrkVeto and PassTauVeto
 
-        totlep = countEle + countMu + countIsk
+        totlep = countEle + countMu
         PassLLLep = (totlep == 1) and sum([ e.MtW for e in electrons if e.Stop0l ] + 
-                                          [ m.MtW for m in muons if m.Stop0l ] + 
-                                          [ i.MtW for i in isotracks if i.Stop0l ]) < 100
+                                          [ m.MtW for m in muons if m.Stop0l ]) < 100
 
         PassNjets       = self.PassNjets(jets)
         PassMET         = met.pt >= 250
         PassHT          = stop0l.HT >= 300
         ## In case JEC changed jet pt order, resort jets
         sortedPhi = self.GetJetSortedIdx(jets)
         PassdPhiLowDM   = self.PassdPhi(sortedPhi, [0.5, 0.15, 0.15])
+	PassdPhiMedDM   = self.PassdPhi(sortedPhi, [0.15, 0.15, 0.15], invertdPhi=True) #Variable for LowDM Validation bins
```

Comment:
This is not the MedDM, it is just inverted dPhiLowDM.

Reply:
This is not the same as dPhiLowDM. This is min(dphi(j123,MET)) < 0.15 while dPhiLowDM is dphi(j1,MET) > 0.5 && dphi(j12,MET) > 0.15. For the new dPhiMedDM is only used for the last four bins in the LM Validation region as !Pass_dPhiMET && !Pass_dPhiMETMedDM as per Hui's definition.

Reply:
I see. So this has to combined with !Pass_dPhiMET && !Pass_dPhiMETMedDM  for it to work? Sounds danger. @wang-hui Can you please provide the exact definition and we should implement then here.

Reply:
Hi @zhenbinwu and @mkilpatr 
For the last 4 bins in LM Validation, we require !min(dPhi12(3)) < 0.15 to avoid too many QCD events.
At the same time, we require !Pass_dPhiMET to avoid overlaping with LM search region.
So the exact defination is (!min(dPhi12(3)) < 0.15) && (!Pass_dPhiMET)
I just have a discussion with Zhenbin and we both agree that writing down the dPhi range for each jet explicitly is more straitforward.
For LM Validation, it is 0.15 < dPhi1 < 0.5 && dPhi2 > 0.15 and dPhi3 (if exists) > 0.15.
You can chekc that it is identical to (!min(dPhi12(3)) < 0.15) && (!Pass_dPhiMET).

For HM Validation, we require Pass_dPhiMET so that the validation region is close to the search region.
At the same time, we require !Pass_dPhiHighMET to avoid overlaping with HM search region.
So the exact defination is Pass_dPhiMET && !Pass_dPhiHighMET

Reply:
Ok, I would prefer clear and exact logic in the meddPhi definition. Matt can you update it? Otherwise I can work on it once I merge this PR.

Reply:
Hey @wang-hui from your past emails and talks on the validation the definition of the LM Validation for the last 4 bins should be !Pass_dPhiMET && !Pass_dPhiMETMedDM. Above you wrote Pass_dPhiMET && !Pass_dPhiHighMET which is not the definition. Also @zhenbinwu I have already confirmed that this variable is the correct definition of the LM validation region

Reply:
Hi @mkilpatr I listed both LM dPhi and HM dPhi for validation. Could you please read again?

Reply:
Ah well the HM is already in agreement as per your slides. I am also using Pass_dPhiMET && !Pass_dPhiHighMET. We have also confirmed that the LM bins are the same which @wang-hui, @caleb-james-smith and I discussed last week. So !Pass_dPhiMET && !Pass_dPhiMETMedDM is correct for the last four bins on the LM validation